### PR TITLE
Implement Nondetbigint3v1 hint

### PR DESCRIPTION
**Pull request by har777**
Implements #265

### CODE REVIEW

File: pkg/hintrunner/zero/zerohint_ec.go [hunk 1]
```diff
@@ -95,3 +96,82 @@ func createEcNegateHinter(resolver hintReferenceResolver) (hinter.Hinter, error)
...
+			//?     a = []
+			//>     for _ in range(3):
+			//>         num, residue = divmod(num, BASE)
+			//>         a.append(residue)
+			//>     assert num == 0
+			//>     return a
+			//>
+			//> segments.write_arg(ids.res.address_, split(value))
+
+			var residue big.Int
```
Comment (MaksymMalicki) [inline pkg/hintrunner/zero/zerohint_ec.go:141]: we should move those functionalities to the separate secp utils file. Maybe the generation of the constants as well?

File: pkg/hintrunner/zero/zerohint_ec_test.go [hunk 2]
```diff
@@ -180,5 +182,240 @@ func TestZeroHintEc(t *testing.T) {
...
+
+					err := ctx.ScopeManager.AssignVariable("value", value)
+					if err != nil {
+						t.Errorf("Error assigning variable value in scope")
+					}
+				},
+				makeHinter: func(ctx *hintTestContext) hinter.Hinter {
+					return newNondetBigint3V1Hint(ctx.operanders["res"])
+				},
+				check: consecutiveVarValueEquals("res", []*fp.Element{feltString("123456"), feltString("0"), feltString("0")}),
```
Comment (MaksymMalicki) [inline pkg/hintrunner/zero/zerohint_ec_test.go:247]: maybe use FeltZero from consts here? I see more cases where you have used feltString("0")

File: pkg/hintrunner/zero/zerohint_ec_test.go [hunk 3]
```diff
@@ -180,5 +182,240 @@ func TestZeroHintEc(t *testing.T) {
...
+
+					err := ctx.ScopeManager.AssignVariable("value", value)
+					if err != nil {
+						t.Errorf("Error assigning variable value in scope")
+					}
+				},
+				makeHinter: func(ctx *hintTestContext) hinter.Hinter {
+					return newNondetBigint3V1Hint(ctx.operanders["res"])
+				},
+				check: consecutiveVarValueEquals("res", []*fp.Element{feltString("1"), feltString("1"), feltString("0")}),
```
Comment (MaksymMalicki) [inline pkg/hintrunner/zero/zerohint_ec_test.go:332]: Maybe use const FeltOne here?

File: pkg/hintrunner/zero/zerohint_ec.go (hunk 1 above)
Comment (MaksymMalicki) [reply pkg/hintrunner/zero/zerohint_ec.go:141]: I mean the secp constants

Comment (har777) [reply pkg/hintrunner/zero/zerohint_ec.go:141]: Yeah lots of functionality is clumped in 1 utils file right now. We need separate files for math utils, secp utils, etc.

Comment (har777) [reply pkg/hintrunner/zero/zerohint_ec.go:141]: I think its cleaner to just write a `split` function and use it here. I'll do that 👍🏽

Comment (har777) [reply pkg/hintrunner/zero/zerohint_ec.go]: Done! Please check.

**Review by MaksymMalicki — APPROVED** [review]

File: pkg/hintrunner/zero/zerohint_ec.go (hunk 1 above)
Comment (MaksymMalicki) [reply pkg/hintrunner/zero/zerohint_ec.go]: Looks good!

File: pkg/hintrunner/zero/zerohint_ec_test.go [hunk 4]
```diff
@@ -181,5 +183,240 @@ func TestZeroHintEc(t *testing.T) {
...
+					value, ok := new(big.Int).SetString("3618502788666127798953978732740734578953660990361066340291730267696802036752", 10)
+					if !ok {
+						t.Errorf("Error creating big.Int")
+					}
+
+					err := ctx.ScopeManager.AssignVariable("value", value)
+					if err != nil {
+						t.Errorf("Error assigning variable value in scope")
+					}
+				},
```
Comment (cicr99) [inline pkg/hintrunner/zero/zerohint_ec_test.go:225]: As you've used this logic several times in these tests, I believe you could create a function with it as to not have to repeat it every time

File: pkg/hintrunner/zero/zerohint_test.go [hunk 5]
```diff
@@ -279,10 +279,14 @@ func runHinterTests(t *testing.T, tests map[string][]hintTestCase) {
...
 				}
-				if _, ok := set[string(stringKey)]; ok {
-					t.Fatalf("%s: duplicated test case case (i=%d) found: %s", testGroup, i, stringKey)
+
+				// TODO: temporary hack, figure out a way to make sure tests dont get marked as duplicate even when the scope they start with is different
+				if tc.ctxInit == nil {
+					if _, ok := set[string(stringKey)]; ok {
+						t.Fatalf("%s: duplicated test case case (i=%d) found: %s", testGroup, i, stringKey)
+					}
+					set[string(stringKey)] = struct{}{}
```
Comment (cicr99) [inline pkg/hintrunner/zero/zerohint_test.go:288]: You can remove this sanity check altogether, it's not sustainable.

File: pkg/hintrunner/zero/zerohint_ec_test.go (hunk 4 above)
Comment (har777) [reply pkg/hintrunner/zero/zerohint_ec_test.go]: Changed to the following format:
```
value := bigIntString("154742504910672534362390530", 10)
ctx.ScopeManager.EnterScope(map[string]any{"value": value})
```

**Review by cicr99 — APPROVED** [review]

**Pull request merged**